web/src/components/merchant/sign-up/SignUpCard.tsx

```diff
@@ -56,31 +63,142 @@ const StyledButton = styled(Button)`
   font-size: 18px;
   font-weight: bolder;
   text-transform: uppercase;
-
-  margin-top: 30px;
+  margin: 0;
 `;
 
+interface SignUpState {
+  name?: string;
+  email?: string;
+  password?: string;
+  confirmPassword?: string;
+  vpa?: string;
+  errorMessage?: string;
+}
+
 /**
  * This is the card containing the sign up form for the Merchant app.
  * It is displayed in the Sign Up page.
  */
-const SignUpCard: React.FC = () => (
-  <CardContainer>
-    <GroupBuyMerchantHeader />
-    <Form>
-      <FormRow label="Name" inputType="text" />
-      <FormRow label="Email" inputType="email" />
-      <FormRow label="Password" inputType="password" />
-      <FormRow label="Confirm Password" inputType="password" />
-      <FormRow label="VPA" inputType="email" />
-      <StyledRow>
-        <StyledButton>Sign Up</StyledButton>
-      </StyledRow>
-    </Form>
-    <div>
-      Already have an account? <Link to="/merchant/sign-in">Sign in</Link> now!
-    </div>
-  </CardContainer>
-);
+@autobind
+class SignUpCard extends React.Component<{}, SignUpState> {
+  constructor(props: Readonly<{}>) {
+    super(props);
+    this.state = {
+      name: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+      vpa: '',
+      errorMessage: '',
+    };
+  }
+
+  handleInputChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const name: string = event.target.name;
+    const value: string = event.target.value;
+
+    this.setState({
+      [name]: value,
+      errorMessage: '',
+    });
+  }
+
+  handleSubmit() {
+    const name: string | undefined = this.state.name;
+    const email: string | undefined = this.state.email;
+    const password: string | undefined = this.state.password;
+    const confirmPassword: string | undefined = this.state.confirmPassword;
+    const vpa: string | undefined = this.state.vpa;
+
+    if (!(name && email && password && confirmPassword && vpa)) {
+      this.setState({
+        errorMessage: 'Please fill in all of the fields.',
+      });
+      return;
+    }
+
+    if (password !== confirmPassword) {
+      this.setState({
+        errorMessage: 'Passwords do not match.',
+      });
+      return;
+    }
+
+    // Add a new user account to Firebase.
```

Comment:
1. Why do we need this?
2. Can we just call server after submission, and move this logic to server side?

Reply:
1. We are using Firebase authentication, so next time when user signs in, Firebase can do the authentication for us.
I will add a comment in the code explaining this.

2. We can but since we are sending password to server, we then have to write a code to hash the password ourselves. Here, Firebase client library does that for us so we don't have to do a lot of work. We can just send a token that they give us to the server, and then the server side verify that with the Firebase.

Reply:
I am moving this part to a new PR because I realised I am doing 2 things in this PR.